### ISSUE TITLE
`no-dupe-keys` not working with `<script setup>`

### ISSUE THREAD

**Issue by sapphi-red** [vuejs/eslint-plugin-vue#1837]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.12.0
- **eslint-plugin-vue version:** 8.5.0
- **Node version:** 16.14.2
- **Operating System:** Windows

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  extends: [
    'plugin:vue/base'
  ],
  parserOptions: {
    parser: '@typescript-eslint/parser'
  },
  rules: {
    'vue/no-dupe-keys': 'error'
  }
}
```

**What did you do?**	
```vue
<template>
  <div></div>
</template>

<script setup>
import { ref, computed } from "vue"

const props = defineProps({
  a: String,
  b: Number
})

const a = computed(() => props.a + ':')

const b = 3
</script>
```
```vue
<template>
  <div></div>
</template>

<script lang="ts" setup>
import { ref, computed } from "vue"

const props = defineProps<{
  a: string
  b: number
}>()

const a = computed(() => props.a + ':')

const b = 3
</script>
```

**What did you expect to happen?**
Both of the above script will report the following errrors.

- `Duplicated key 'a'`
- `Duplicated key 'b'`

**What actually happened?**
No error was reported.

**Repository to reproduce this issue**
https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxNjs0KwjAQhF9lyMWKrQW9hdpHEMFrLm2zlaD5IT+ClL67CSJ6WZjdmZ1vYZOVxDjrImn3GCL1wgCdVM++a8sUpmv/blmGySsXESgmlzdKO+sjFniaa0xWuxRJYsXsrYZgz0SCleBkTYhw3rqAEyTNytClqGopnQPHNXplbnVRI8c56ZG8MOv2lx5y8ltRVVuc+s/D/YAdNnzzZx2z9VjoP7yZlNXMpwcFxpdC1RrbyOSoudMr7w7r+gbr81zH

https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxNUEFqxDAM/IrQZbM0aaC9GSfsE/oAX5xYWUwT29jywhLy99qkpXsRjEajGWnH2RtCgZJpC6tmGpUDkMY+RtnXqpzsX7gC0xxtYFi1uw8KOSmERJxDYe0WfGTYIdLSwuy3kJkMHLBEv4HCRyaFdcnsXWII0YcEAxharKOviuRe/bWAxNG6ewWTAJe3iaJyx9hc/+W6SP88muYKw3hufNfwBhdxeRmdyuhnPeUMX6JiizGvlFDsNVbvfGdyoO6bnqX3cbQYdEwUy29u/Ax0CjtKq3Xc/3LHD7aTbsE=

**Comment by sapphi-red** [comment]
Because Vue does not show any warning about this usage, I reconsidered that it is intended to not include this in `no-dupe-keys`.
